File: Graduating-Server/src/main/kotlin/com/bestswlkh0310/graduating/graduatingserver/core/school/SchoolType.kt
package com.bestswlkh0310.graduating.graduatingserver.core.school

enum class SchoolType(
    val limit: Int
) {
    HIGH(limit = 3),
    MIDDLE(limit = 3),
    ELEMENTARY(limit = 6);

    companion object {
        fun ofKorean(string: String) = entries.firstOrNull { it.korean() == string }
    }

    fun korean() = when(this) {
        HIGH -> "고등학교"
        MIDDLE -> "중학교"
        ELEMENTARY -> "초등학교"
    }
}

File: Graduating-iOS/Shared/Model/School/SchoolType.swift
public enum SchoolType: String, ModelProtocol {
    case high = "HIGH"
    case middle = "MIDDLE"
    case elementary = "ELEMENTARY"
}

public extension SchoolType {
    var limit: Int {
        switch self {
        case .high:
            3
        case .middle:
            3
        case .elementary:
            6
        }
    }
}
